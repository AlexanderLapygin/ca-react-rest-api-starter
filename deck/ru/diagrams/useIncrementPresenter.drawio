<mxfile host="Electron" modified="2020-06-21T08:56:55.301Z" agent="5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/13.2.4 Chrome/83.0.4103.104 Electron/9.0.4 Safari/537.36" etag="P7bQZgmek9JXQYm7Yk17" version="13.2.4" type="device"><diagram name="Page-1" id="b5b7bab2-c9e2-2cf4-8b2a-24fd1a2a6d21">7Vvbdto4FP0aHvHSxbLsxwBJm3Y6Jc2kTfrmgAJujUVtESBfP5Jt2fKFSwjM6nTB6iLWltCxj/bZOrLUDu7PVu9ifz79xMcs7CAwXnXwoIOQjYgn/yhknSEIejRDJnEwzjBYArfBC8tBkKOLYMySSkPBeSiCeRUc8ShiI1HB/Djmy2qzJx5Wrc79CWsAtyM/bKLfgrGYZqiLaIm/Z8Fkqi1DJ3/iR3/0cxLzRZTb6yD8lH6y6pmv+8ofNJn6Y740IHzZwf2Yc5FdzVZ9Firnardlv7vaUFvcd8wisc8PRvHgy2MYgNW34fK7d+cOvQ+rLnKzbp79cMH0c6R3K9baQ+kzMtUL6ODechoIdjv3R6p2KUkhsamYhbIE5eVTEIZ9HvI4/S0GFDqwr3AeiXzwMcjLRrur9CPxRMT8JzN76EuWyQfqjf1kmt6FMtN8+NwfzywWbGVAuTPeMT5jIl7LJnltF9lOPjQ5d7sE5MCyZAIEOAenJg2Aburn/JsUBsoxkBf5MLQPyZ2zZD6ht8PB9Gbx4P+4WrhJF+8eEUW7+f4uKILHf9Q9gB2uwZRWXVPQeGkGSYtnbPcIjvl2dzO9fuj5xIbufPUZXaztD29wzFY/v9lbLnQsz8E2toHUP0JxxXMIuhYAUIWC48lv12lSDFsAO8QhHqXURrZNm27F0PI8SrBrO1B9E/dEXoa7vSyVbK4uY/ZrEcRsfB0JFj+limBKgfJsIIX2L/+RhUOeBCLgkax75ELwWVus11RA67Fb0w/c1A/c7/UuBxKPufBzO13oglap2Mqu3frxxpGjJ9IN9LuFh+En6lLLhScSC+flfXI3YvHkF1yI+x6I/FXUhW08dkKRU6fiF+fXguuKbpKS7EI2IM58VVbKq4n6u0jYdTSK2Uy6bxizhCny664fY91MI/LmM3sarg2JES/jmM//8eMJE/lUu2984Oa8i5EHerW4QS1xU/QY1oJ0lD3X5igWXM38fhhMWptf5BVFsO9JO7SBdpvVF1uF8hICnaa8Og62IHVsl8p/iNL/lod7hKXW0xGfzXmU+qf3gwmxNgUwBYpM1G716PZI2NulsOHCtoleYzELpeI+V9PpNj/m9oY8SOOvyDWIZQPP+FQzDzm5EgcYH1S1wJ+eEhkx9XEqnmqvoYsuP4mb4Ys/n/R/XCw+Dj87QLTOhFlQJ3M/2qwgT/4sCNeZhsx4xJM0WUbZXARqIgPaNKaDSH3JReSNKzRdLxQl/SAkfRSJDNS1Ghmi7pZIl+xqC4u2mkoHdYPKbjKHFTVlReavokKvKxSgTOEej8WUT3gk1YZLeVEVMK1IyZ/KmML8heApnIpn2YpF4wu1IkyhSEaSRq+CMDTMFJqoILvEMl3U96e1UdWaSry5PgvXtNPM8ioQ94ZdVX7IyhYpkMGq1mSwNoDSe2w8MZy3ZdiKqmLEyhAnRry29SWjqoAnxq8q3eURTFQMF627MnItBxpxXPa6NmxRaDlmPLvtxhO+iEessNRkYftNQIA22UXthkQ64240lBYrHqxW5MFggtUQ1R3UY7lIDzIxqWYNZ4E5C8xZYM4CcwKBOeLLvWKR8ZqFyqnWARSj2nsXaLmkkcnaGxaitoW3pK77LgFa80jnnEeeZf4s82eZP+eRZ4E5C8xZYP4Igflj80hQSyORBZvvlG3XQvhkaaS4//Q16V/xmyv48fvifjabLS+6sC2P3DoKNZXKfF8qVK5PlZEplCnXpVKVsi4P3sbfMJLqPXamPln/mfJkulOoTlk1WOeF141+FqO7U/RUubZx5K3vuqFDPMtzbQggtV0XUFJ91w2o5A8u94qrBipSg4yTHjUzqEZhrFgJiw1Ip7YDWRGWslvJA39tNJurBskbX7S3jpK3a6tuVPCoTECKwzYltHtLz27PYiSlLoD67pH0O7tOcalL6ptqHHZcR+Py+mrffb5X6ePm6KkpJ+rTvnd5QjWUXLL0jktOJzl7ao0zBVGf56psEWkxPfpmN7R3i2FTy/ZzsbFjChoCWJUvaMgXfIV8yVGK1/dmwfiVKpY/S0unlT2hN513BOhOeTT4QFrooLE3qii0aydsMD5QJ2GjIzmrb95q3CCUB8ggiMhzcPtVwJ838c3s4fvt3y/+PidvzpQ2KL2Nqialt23u/i6URm4r8zQx07ySlNXkQMJjqebmysdtmjG34mtmjsf/1hNYenL6TfJb+xj5LT00mmRhyOJA+lQdq6lGGDxJhOFmhG07IHi0CHtj6Mh0xCOS2ATZjo2QW6c0qp6+PHCqoB6WZogLIbSphxy7kWFLKy4hgMrbwMQ+VeC0jlzbGe7T5M+Iyvw5TZxrKfQX5o/EvrmwzD3Ftqkrj0wz182hncfNZsF4rMy0ZthV9TAnRarL+U2CZswccr4cVhdiXf3KwJhUPNSSNp8sa955KPJoTIG0baXVlyMgB644QHkd/W85A2mrtL6eJlCuyqs0sVv+FwI+Dk86+Vs/Q3/K93348l8=</diagram></mxfile>