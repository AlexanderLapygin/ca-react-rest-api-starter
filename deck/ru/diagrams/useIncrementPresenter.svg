<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" width="1031px" height="1001px" viewBox="-0.5 -0.5 1031 1001" content="&lt;mxfile host=&quot;Electron&quot; modified=&quot;2020-06-21T08:56:59.054Z&quot; agent=&quot;5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/13.2.4 Chrome/83.0.4103.104 Electron/9.0.4 Safari/537.36&quot; etag=&quot;5Z4uDPSBDcVo00wYXem-&quot; version=&quot;13.2.4&quot; type=&quot;device&quot;&gt;&lt;diagram name=&quot;Page-1&quot; id=&quot;b5b7bab2-c9e2-2cf4-8b2a-24fd1a2a6d21&quot;&gt;7Vvbdto4FP0aHvHSxbLsxwBJm3Y6Jc2kTfrmgAJujUVtESBfP5Jt2fKFSwjM6nTB6iLWltCxj/bZOrLUDu7PVu9ifz79xMcs7CAwXnXwoIOQjYgn/yhknSEIejRDJnEwzjBYArfBC8tBkKOLYMySSkPBeSiCeRUc8ShiI1HB/Djmy2qzJx5Wrc79CWsAtyM/bKLfgrGYZqiLaIm/Z8Fkqi1DJ3/iR3/0cxLzRZTb6yD8lH6y6pmv+8ofNJn6Y740IHzZwf2Yc5FdzVZ9Firnardlv7vaUFvcd8wisc8PRvHgy2MYgNW34fK7d+cOvQ+rLnKzbp79cMH0c6R3K9baQ+kzMtUL6ODechoIdjv3R6p2KUkhsamYhbIE5eVTEIZ9HvI4/S0GFDqwr3AeiXzwMcjLRrur9CPxRMT8JzN76EuWyQfqjf1kmt6FMtN8+NwfzywWbGVAuTPeMT5jIl7LJnltF9lOPjQ5d7sE5MCyZAIEOAenJg2Aburn/JsUBsoxkBf5MLQPyZ2zZD6ht8PB9Gbx4P+4WrhJF+8eEUW7+f4uKILHf9Q9gB2uwZRWXVPQeGkGSYtnbPcIjvl2dzO9fuj5xIbufPUZXaztD29wzFY/v9lbLnQsz8E2toHUP0JxxXMIuhYAUIWC48lv12lSDFsAO8QhHqXURrZNm27F0PI8SrBrO1B9E/dEXoa7vSyVbK4uY/ZrEcRsfB0JFj+limBKgfJsIIX2L/+RhUOeBCLgkax75ELwWVus11RA67Fb0w/c1A/c7/UuBxKPufBzO13oglap2Mqu3frxxpGjJ9IN9LuFh+En6lLLhScSC+flfXI3YvHkF1yI+x6I/FXUhW08dkKRU6fiF+fXguuKbpKS7EI2IM58VVbKq4n6u0jYdTSK2Uy6bxizhCny664fY91MI/LmM3sarg2JES/jmM//8eMJE/lUu2984Oa8i5EHerW4QS1xU/QY1oJ0lD3X5igWXM38fhhMWptf5BVFsO9JO7SBdpvVF1uF8hICnaa8Og62IHVsl8p/iNL/lod7hKXW0xGfzXmU+qf3gwmxNgUwBYpM1G716PZI2NulsOHCtoleYzELpeI+V9PpNj/m9oY8SOOvyDWIZQPP+FQzDzm5EgcYH1S1wJ+eEhkx9XEqnmqvoYsuP4mb4Ys/n/R/XCw+Dj87QLTOhFlQJ3M/2qwgT/4sCNeZhsx4xJM0WUbZXARqIgPaNKaDSH3JReSNKzRdLxQl/SAkfRSJDNS1Ghmi7pZIl+xqC4u2mkoHdYPKbjKHFTVlReavokKvKxSgTOEej8WUT3gk1YZLeVEVMK1IyZ/KmML8heApnIpn2YpF4wu1IkyhSEaSRq+CMDTMFJqoILvEMl3U96e1UdWaSry5PgvXtNPM8ioQ94ZdVX7IyhYpkMGq1mSwNoDSe2w8MZy3ZdiKqmLEyhAnRry29SWjqoAnxq8q3eURTFQMF627MnItBxpxXPa6NmxRaDlmPLvtxhO+iEessNRkYftNQIA22UXthkQ64240lBYrHqxW5MFggtUQ1R3UY7lIDzIxqWYNZ4E5C8xZYM4CcwKBOeLLvWKR8ZqFyqnWARSj2nsXaLmkkcnaGxaitoW3pK77LgFa80jnnEeeZf4s82eZP+eRZ4E5C8xZYP4Igflj80hQSyORBZvvlG3XQvhkaaS4//Q16V/xmyv48fvifjabLS+6sC2P3DoKNZXKfF8qVK5PlZEplCnXpVKVsi4P3sbfMJLqPXamPln/mfJkulOoTlk1WOeF141+FqO7U/RUubZx5K3vuqFDPMtzbQggtV0XUFJ91w2o5A8u94qrBipSg4yTHjUzqEZhrFgJiw1Ip7YDWRGWslvJA39tNJurBskbX7S3jpK3a6tuVPCoTECKwzYltHtLz27PYiSlLoD67pH0O7tOcalL6ptqHHZcR+Py+mrffb5X6ePm6KkpJ+rTvnd5QjWUXLL0jktOJzl7ao0zBVGf56psEWkxPfpmN7R3i2FTy/ZzsbFjChoCWJUvaMgXfIV8yVGK1/dmwfiVKpY/S0unlT2hN513BOhOeTT4QFrooLE3qii0aydsMD5QJ2GjIzmrb95q3CCUB8ggiMhzcPtVwJ838c3s4fvt3y/+PidvzpQ2KL2Nqialt23u/i6URm4r8zQx07ySlNXkQMJjqebmysdtmjG34mtmjsf/1hNYenL6TfJb+xj5LT00mmRhyOJA+lQdq6lGGDxJhOFmhG07IHi0CHtj6Mh0xCOS2ATZjo2QW6c0qp6+PHCqoB6WZogLIbSphxy7kWFLKy4hgMrbwMQ+VeC0jlzbGe7T5M+Iyvw5TZxrKfQX5o/EvrmwzD3Ftqkrj0wz182hncfNZsF4rMy0ZthV9TAnRarL+U2CZswccr4cVhdiXf3KwJhUPNSSNp8sa955KPJoTIG0baXVlyMgB644QHkd/W85A2mrtL6eJlCuyqs0sVv+FwI+Dk86+Vs/Q3/K93348l8=&lt;/diagram&gt;&lt;/mxfile&gt;" style="background-color: rgb(255, 255, 255);"><defs/><g><rect x="0" y="0" width="1030" height="1000" fill="#07161c" stroke="#0c242e" stroke-dasharray="3 3" pointer-events="all"/><path d="M 899.96 498 Q 913 498 913 514 Q 913 530 899.96 530" fill="none" stroke="#ffffff" stroke-width="8" stroke-miterlimit="10" transform="rotate(-180,906.48,514)" pointer-events="all"/><rect x="166.43" y="280" width="663.18" height="480" fill="#3290b8" stroke="#ffffff" stroke-width="3" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe flex-end; justify-content: unsafe center; width: 1px; height: 1px; padding-top: 277px; margin-left: 498px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 23px; font-family: Helvetica; color: #FFFFFF; line-height: 1.2; pointer-events: all; white-space: nowrap; "><font style="font-size: 56px">useIncrementPresenter<br /></font></div></div></div></foreignObject><text x="498" y="277" fill="#FFFFFF" font-family="Helvetica" font-size="23px" text-anchor="middle">useIncrementPresenter&#xa;</text></switch></g><path d="M 808.2 296.56 L 824.2 296.56 L 824.2 316.56 L 808.2 316.56 L 808.2 312.56 L 804.2 312.56 L 804.2 308.56 L 808.2 308.56 L 808.2 304.56 L 804.2 304.56 L 804.2 300.56 L 808.2 300.56 Z" fill="#ffffff" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><path d="M 808.2 300.56 L 812.2 300.56 L 812.2 304.56 L 808.2 304.56 M 808.2 308.56 L 812.2 308.56 L 812.2 312.56 L 808.2 312.56" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><rect x="815" y="491.85" width="47.81" height="44.31" fill="#3290b8" stroke="#ffffff" stroke-width="3" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 46px; height: 1px; padding-top: 514px; margin-left: 816px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><span style="font-family: monospace ; font-size: 0px">%3CmxGraphModel%3E%3Croot%3E%3CmxCell%20id%3D%220%22%2F%3E%3CmxCell%20id%3D%221%22%20parent%3D%220%22%2F%3E%3CmxCell%20id%3D%222%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BendArrow%3Dnone%3BendFill%3D0%3BfontSize%3D40%3BfontColor%3D%23FFFFFF%3BstrokeColor%3D%23FFFFFF%3BstrokeWidth%3D4%3BexitX%3D0%3BexitY%3D0.5%3BexitDx%3D0%3BexitDy%3D0%3B%22%20edge%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%3E%3CmxPoint%20x%3D%22-999.6199999999999%22%20y%3D%22171.66000000000008%22%20as%3D%22sourcePoint%22%2F%3E%3CmxPoint%20x%3D%22-1029%22%20y%3D%22172%22%20as%3D%22targetPoint%22%2F%3E%3C%2FmxGeometry%3E%3C%2FmxCell%3E%3C%2Froot%3E%3C%2FmxGraphModel%3E</span><span style="font-family: monospace ; font-size: 0px">%3CmxGraphModel%3E%3Croot%3E%3CmxCell%20id%3D%220%22%2F%3E%3CmxCell%20id%3D%221%22%20parent%3D%220%22%2F%3E%3CmxCell%20id%3D%222%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BendArrow%3Dnone%3BendFill%3D0%3BfontSize%3D40%3BfontColor%3D%23FFFFFF%3BstrokeColor%3D%23FFFFFF%3BstrokeWidth%3D4%3BexitX%3D0%3BexitY%3D0.5%3BexitDx%3D0%3BexitDy%3D0%3B%22%20edge%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%3E%3CmxPoint%20x%3D%22-999.6199999999999%22%20y%3D%22171.66000000000008%22%20as%3D%22sourcePoint%22%2F%3E%3CmxPoint%20x%3D%22-1029%22%20y%3D%22172%22%20as%3D%22targetPoint%22%2F%3E%3C%2FmxGeometry%3E%3C%2FmxCell%3E%3C%2Froot%3E%3C%2FmxGraphModel%3E</span></div></div></div></foreignObject><text x="839" y="518" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">%3CmxGra...</text></switch></g><rect x="153" y="492.16" width="48.23" height="44.31" fill="#3290b8" stroke="#ffffff" stroke-width="3" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 46px; height: 1px; padding-top: 514px; margin-left: 154px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><span style="font-family: monospace ; font-size: 0px">%3CmxGraphModel%3E%3Croot%3E%3CmxCell%20id%3D%220%22%2F%3E%3CmxCell%20id%3D%221%22%20parent%3D%220%22%2F%3E%3CmxCell%20id%3D%222%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BendArrow%3Dnone%3BendFill%3D0%3BfontSize%3D40%3BfontColor%3D%23FFFFFF%3BstrokeColor%3D%23FFFFFF%3BstrokeWidth%3D4%3BexitX%3D0%3BexitY%3D0.5%3BexitDx%3D0%3BexitDy%3D0%3B%22%20edge%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%3E%3CmxPoint%20x%3D%22-999.6199999999999%22%20y%3D%22171.66000000000008%22%20as%3D%22sourcePoint%22%2F%3E%3CmxPoint%20x%3D%22-1029%22%20y%3D%22172%22%20as%3D%22targetPoint%22%2F%3E%3C%2FmxGeometry%3E%3C%2FmxCell%3E%3C%2Froot%3E%3C%2FmxGraphModel%3E</span><span style="font-family: monospace ; font-size: 0px">%3CmxGraphModel%3E%3Croot%3E%3CmxCell%20id%3D%220%22%2F%3E%3CmxCell%20id%3D%221%22%20parent%3D%220%22%2F%3E%3CmxCell%20id%3D%222%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BendArrow%3Dnone%3BendFill%3D0%3BfontSize%3D40%3BfontColor%3D%23FFFFFF%3BstrokeColor%3D%23FFFFFF%3BstrokeWidth%3D4%3BexitX%3D0%3BexitY%3D0.5%3BexitDx%3D0%3BexitDy%3D0%3B%22%20edge%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%3E%3CmxPoint%20x%3D%22-999.6199999999999%22%20y%3D%22171.66000000000008%22%20as%3D%22sourcePoint%22%2F%3E%3CmxPoint%20x%3D%22-1029%22%20y%3D%22172%22%20as%3D%22targetPoint%22%2F%3E%3C%2FmxGeometry%3E%3C%2FmxCell%3E%3C%2Froot%3E%3C%2FmxGraphModel%3E</span></div></div></div></foreignObject><text x="177" y="518" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">%3CmxGra...</text></switch></g><path d="M 153 514.31 L 103 514.31" fill="none" stroke="#ffffff" stroke-width="4" stroke-miterlimit="10" pointer-events="stroke"/><rect x="303.02" y="379.31" width="410" height="270" fill="#2c7c9e" stroke="#ffffff" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 408px; height: 1px; padding-top: 514px; margin-left: 304px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font color="#ffffff" style="font-size: 40px">Реализация</font></div></div></div></foreignObject><text x="508" y="518" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">Реализация</text></switch></g><path d="M 201.23 514.31 L 303.02 514.31" fill="none" stroke="#ffffff" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 713.02 514.31 L 815 514" fill="none" stroke="#ffffff" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 899.96 514 L 862.81 514" fill="none" stroke="#ffffff" stroke-width="7" stroke-miterlimit="10" pointer-events="stroke"/><rect x="50" y="477" width="92" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 90px; height: 1px; padding-top: 487px; margin-left: 51px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 27px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font color="#ffffff" style="font-size: 27px">React</font></div></div></div></foreignObject><text x="96" y="495" fill="#000000" font-family="Helvetica" font-size="27px" text-anchor="middle">React</text></switch></g><rect x="845" y="460" width="132" height="20" fill="none" stroke="none" pointer-events="all"/><g transform="translate(-0.5 -0.5)"><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 130px; height: 1px; padding-top: 470px; margin-left: 846px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 17px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><font color="#ffffff" style="font-size: 17px">CounterIncrementIn</font></div></div></div></foreignObject><text x="911" y="475" fill="#000000" font-family="Helvetica" font-size="17px" text-anchor="middle">CounterIncrement...</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://desk.draw.io/support/solutions/articles/16000042487" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>