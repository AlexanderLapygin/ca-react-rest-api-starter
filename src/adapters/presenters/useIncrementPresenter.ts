import { CounterIncrementIn } from '../../domain/usecases'
import { useCallback, useEffect, useState } from 'react'

export interface IncrementPresenterAPI {
  increment(): Promise<void>
  updateUI(counter: number): void
}

const useIncrementPresenter: (
  counterIncrementIn: CounterIncrementIn
) => [{ counter: number }, IncrementPresenterAPI] = (
  counterIncrementIn: CounterIncrementIn
) => {
  const [counter, setCounter] = useState<number>(0)

  useEffect(() => {
    ;(async (): Promise<void> => {
      try {
        const newCounter: number = await counterIncrementIn.getCounter()
        setCounter(newCounter)
      } catch (error) {
        console.error(error)
      }
    })()
  }, [counterIncrementIn])

  const handleIncrement = useCallback(async (): Promise<void> => {
    try {
      const newCounter: number = await counterIncrementIn.increment()
      setCounter(newCounter)
    } catch (error) {
      console.error(error)
    }
  }, [counterIncrementIn])

  const state = {
    counter
  }

  const api: IncrementPresenterAPI = {
    increment: handleIncrement,
    updateUI: setCounter
  }

  return [state, api]
}

export default useIncrementPresenter
