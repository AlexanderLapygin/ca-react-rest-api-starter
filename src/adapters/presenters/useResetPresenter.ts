import { CounterResetIn } from '../../domain/usecases/CounterReset'
import { useCallback } from 'react'

export interface ResetPresenterAPI {
  reset(): Promise<void>
}

const useResetPresenter: (
  counterResetIn: CounterResetIn,
  updateUI: any
) => ResetPresenterAPI = (counterResetIn: CounterResetIn, updateUI: any) => {
  const handleReset = useCallback(async (): Promise<void> => {
    try {
      await counterResetIn.reset()
      updateUI(0)
    } catch (error) {
      console.error(error)
    }
  }, [counterResetIn, updateUI])

  const api: ResetPresenterAPI = {
    reset: handleReset
  }

  return api
}

export default useResetPresenter
