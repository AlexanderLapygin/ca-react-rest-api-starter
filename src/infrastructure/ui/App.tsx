import React from 'react'
import Counter from './Counter'
import { AppFactory } from "../../AppFactory";

export const appFactory = new AppFactory()

const App = (): JSX.Element => <Counter />

export default App
