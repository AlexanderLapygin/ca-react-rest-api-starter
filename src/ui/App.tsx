import React from 'react'
import Counter from './Counter'
import { AdapterFactory } from '../adapters'

export const adapterFactory = new AdapterFactory()

const App = (): JSX.Element => <Counter />

export default App
