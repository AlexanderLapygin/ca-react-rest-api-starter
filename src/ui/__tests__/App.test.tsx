import React from 'react'
import { render, cleanup } from '@testing-library/react'
import App from '../App'
import useIncrementPresenter from '../../adapters/presenters/useIncrementPresenter'
import useResetPresenter from '../../adapters/presenters/useResetPresenter'

jest.mock('../../adapters/presenters/useIncrementPresenter')
jest.mock('../../adapters/presenters/useResetPresenter')

describe('App', () => {
  // Mocking Counter hooks to prevent remote fetching
  let incrementMock: any
  let resetMock: any
  let updateUiMock: any

  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    incrementMock = jest.fn().mockImplementation(() => {})
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    resetMock = jest.fn().mockImplementation(() => {})
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    updateUiMock = jest.fn().mockImplementation(() => {})

    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    useIncrementPresenter.mockReturnValue([
      99,
      {
        increment: incrementMock,
        updateUI: updateUiMock
      }
    ])

    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    useResetPresenter.mockReturnValue({ reset: resetMock })
  })

  afterEach(() => {
    incrementMock.mockReset()
    updateUiMock.mockReset()
    resetMock.mockReset()
  })

  it('should match the snapshot', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})
