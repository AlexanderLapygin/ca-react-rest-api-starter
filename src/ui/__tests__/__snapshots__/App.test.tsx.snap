// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="value-testid"
    >
      <button
        data-testid="reset-button-testid"
      >
        Reset
      </button>
       undefined 
      <button
        data-testid="increment-button-testid"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
